### ISSUE TITLE
Cannot init `Cohpcar` in `matgenb ` example notebook: `ValueError: setting an array element with a sequence.`

### ISSUE THREAD

**Issue by DanielYang59** [materialsproject/pymatgen#4215]
The [example in `matgenb` seems to be failing](https://github.com/materialsvirtuallab/matgenb/blob/master/notebooks/2018-03-14-Plotting%20COHP%20from%20LOBSTER.ipynb).

```python
from pymatgen.io.lobster import Cohpcar


COHPCAR_path = "COHPCAR.lobster.txt"
cohpcar = Cohpcar(filename=COHPCAR_path)
```

Gives:
```
Traceback (most recent call last):
  File "/home/yang/developer/pymatgen/debug/recreate_cohp.py", line 5, in <module>
    cohpcar = Cohpcar(filename=COHPCAR_path)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/yang/developer/pymatgen/src/pymatgen/io/lobster/outputs.py", line 138, in __init__
    data = np.array([np.array(line.split(), dtype=float) for line in lines[num_bonds + 3 :]]).transpose()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (402,) + inhomogeneous part.
```

[COHPCAR.lobster.txt](https://github.com/user-attachments/files/18005481/COHPCAR.lobster.txt)

cc @JaGeo @QuantumChemist @naik-aakash

**Comment by JaGeo** [comment]
There is a second notebook. This one is not needed.

**Comment by DanielYang59** [comment]
Thanks for the quick response! This one `2019-01-11-How to plot and evaluate output files from Lobster.ipynb`? This [seems to be failing for the same reason](https://github.com/materialsvirtuallab/matgenb/blob/master/notebooks/2019-01-11-How%20to%20plot%20and%20evaluate%20output%20files%20from%20Lobster.ipynb)

**Comment by JaGeo** [comment]
@naik-aakash could you take a look?

**Comment by JaGeo** [comment]
In general, it would be great if the tutorials would be directly tested on a new pymatgen version. In this way, we have a very hard time to update them.

**Comment by QuantumChemist** [comment]
I'm btw not very actively working on the LOBSTER stuff.

**Comment by naik-aakash** [comment]
Hi @JaGeo , @DanielYang59 , I can look into it 😄

**Comment by DanielYang59** [comment]
@naik-aakash Thanks a lot :)

> I'm btw not very actively working on the LOBSTER stuff.

Sorry I didn't know that, but I would keep that in mind

> In general, it would be great if the tutorials would be directly tested on a new pymatgen version. In the current way, we have a very hard time to update them.

I believe everything would suffer if not checked/maintained regularly. I guess it's good to trigger the test workflow of `matgenb` when `pymatgen` has a new release, but it looks like [we might need a custom trigger for this](https://stackoverflow.com/questions/58465057/trigger-a-github-action-when-another-repository-creates-a-new-release) (i.e. GitHub may not provide native support for this). 

Or perhaps we could do something like https://github.com/materialsproject/pymatgen/blob/31f1e1fb8cc1bb517d59ea8e484965bb641c42a0/.github/workflows/test.yml#L116-L132 

But personally test upon every merge to main sounds too frequent, also I prefer not to modify the release workflow of `pymatgen` as I prefer not to modify an upstream workflow to suit downstream need.

I have collected this into my TODO list :)

**Comment by naik-aakash** [comment]
Hi @DanielYang59 , I had a chance to quickly look into this issue. Seems it not any bug in parser but the example file on the repository has for some reason got added an new line at last line causing this error. Simply removing this trailing line seemed to fix the issue.

Here is an updated file that can be used instead (the only change is for the last line unexpected "\n" is removed, this is usually not the case of files generated by LOBSTER)

[Fixed: COHPCAR.lobster.txt](https://github.com/user-attachments/files/18221909/COHPCAR.lobster.txt)

**Comment by DanielYang59** [comment]
>  Seems it not any bug in parser but the example file on the repository has for some reason got added an new line at last line causing this error. Simply removing this trailing line seemed to fix the issue. For both the example it is same problem.

Hi @naik-aakash wish everything is doing well. That is hugely appreciated. If that's really the case, then LOBSTER might be violating [the POSIX standard definition of "line"](https://pubs.opengroup.org/onlinepubs/9799919799/) IMO:
> 3.185 Line
> A sequence of zero or more non-<newline> characters **plus a terminating <newline> character.**

As per the above definition, every line (including the last) should end with a terminating "newline" char (which might look like an "empty line" at the end of text editor but it's not). 

In other words, a file should always end as `somedata\n` (or other line ending char) instead of `somedata`

Perhaps it's good to reach out to LOBSTER team about this?

**Comment by JaGeo** [comment]
@naik-aakash maybe (after the Christmas break 👀), you can regenerate a cohpcar with the latest Lobster version. Then, we can check this in detail
